### PR TITLE
Sidenav fix

### LINKED ISSUE #472: RTL: Sidenav alignment issues
Screen Shot 2019-08-08 at 11 15 40 AM

Screen Shot 2019-08-08 at 11 18 11 AM

- When you switch to RTL, the icons are cut off when the nav is closed
- There's also no padding between the icons and text when the nav is open, and it kinda looks like there should be some padding between the user icon and the "John doe" info

### LINKED ISSUE #684: [Sidenav] Need Fixed expanded (open) state
### Summary
Building Insights development team needs the component to implement both 2 open states, documented in the [Watson IoT PAL](https://pages.github.ibm.com/Watson-IoT/IoT-Design-Site/components/ui-shell-side-nav/usage) -
* Flexible expanded (open) state - implemented in current component 
* Fixed expanded (open) state - still to be implemented.

**Additional context**
Continuing from https://github.com/IBM/carbon-addons-iot-react/issues/661.
* Current behaviour
1. Go to **Storybook link** - [SideNav Component](https://carbon-addons-iot-react.com/?path=/story/watson-iot-sidenav--sidenav-component).
2. Click on top-left hamburger menu.
3. Side menu opens in Flexible expanded (open) state as an overlay over the main content 

* Desired behaviour
4. Click on top-left hamburger menu.
5. Side menu changes to Fixed expanded (open) state and displaces the main content area to the right.

### Specific timeline issues / requests
Do you want this work within a specific time period? Is it related to an
upcoming release?
* Required by the end of November for next product release.

_NB: The core contributors will try to work with your timeline, but it's not
guaranteed. The earlier you make a request in advance of a desired delivery
date, the better!_

### Available extra resources
What resources do you have to assist this effort? 
Possibly one member of the development team, depending on timelines.
_Carbon Add-ons for Watson IoT is a collaborative system. We encourage teams to build components and submit them for integration._

Same as this one internally in Maximo Asset Monitor:
https://github.ibm.com/wiotp/monitoring-dashboard/issues/785

### LINKED ISSUE #943: [SideNav] isEnabled bit is ignored on each item
Our sidenav code supports an isEnabled prop on each link, but it incorrectly sets a disabled classname (which doesn't do anything)

image

### PR DESCRIPTION
**Pull request by davidicus**
Closes #943, closes #684, closes #472  

**Summary**

This PR takes care of a few issues involving the sidenav.  

**Change List (commits, features, bugs, etc)**

- For #943 I remove the disabled class name that was applied depending on value of the `isEnabled` prop. Recent changes make that class obsolete and was being applied wrong. 
- For #684 I have added a class when the menu toggle is clicked to differentiate from when hovered. This will allow for easier styling of a "content push" layout when side nav is expanded. 
- For #472 there were several issues with RTL. Styles were applied to deal with these issues. 
- I also changed the sidenav story to illustrate how to achieve the "content push" layout. 

**Acceptance Test (how to verify the PR)**

Click on sidenav story. You should see that the story includes a content section that is pushed over when menu is toggled open. This does not happen on hover. 

If you click on the RTL knob you will see the styles adjust to deal with change in orientation.

### CODE REVIEW

**Comment by davidicus** [comment]
@sstone2423 Yeah that is the overlay that is applied on small screens when sidenav is expanded. Our wrapper in storybook is causing this defect. I have added some styles for the story to fix this which actually allowed me to showcase the expanded content state better for RTL. It should all be fixed now! Thanks

File: src/components/SideNav/SideNav.story.scss [hunk 1]
```diff
@@ -0,0 +1,36 @@
+@import '../../globals/vars';
+
+.storybook-container {
+  padding: 0 !important;
+}
+
+.#{$iot-prefix}--main-content {
```
Comment (tay1orjones) [inline src/components/SideNav/SideNav.story.scss]: Do we need to document these styles somewhere to explain how/why we use them for the story? Code comments with rationale would be helpful for this file

Comment (tay1orjones) [reply src/components/SideNav/SideNav.story.scss]: Or should we have two stories for sidenav?
```
▼ Sidenav
 ┣ without push
 ┣ with push
```

Comment (davidicus) [reply src/components/SideNav/SideNav.story.scss]: I have documented the use of "pushing" the content in the info section.

**Comment by davidicus** [comment]
> The overlay renders correctly now, though it's only visible on hover (is that intended?).
> 
> Also a scrollbar appears when the overlay is visible on shorter screen heights. Should the height/placement implementation be changed?
> 
> no hover, no scrollbar	on hover, has scrollbar
> Screen Shot 2020-03-03 at 2 14 08 PM	Screen Shot 2020-03-03 at 2 14 13 PM

this is styling from Carbon. I have added styles to the story to prevent scrollbars

**Pull request merged**